### ISSUE TITLE
Support for Custom SMS Gateways

### ISSUE THREAD

**Issue by grese** [alexkappa/terraform-provider-auth0#416]
### Description

I'd like to request support for `gateway_url`, `provider`, `gateway_authentication`, and `forward_request_info` fields in the `auth0_connection` resource. This would be used to replace Twilio with a custom SMS provider. The changes required are summarized below.

- In the connection's `options` field, the following options would need to be added.
    - `provider` (string; would need to be set to "sms_gateway")
    - `gateway_url` (string; the url of the custom sms gateway)
    - `gateway_authentication` (map; contains the following fields used to authenticate the sms_gateway)
        - `method` (string; would need to be set to "bearer")
        - `subject` (string; subject claim for the HS256 token sent to sms_gateway)
        - `audience` (string; audience claim for the HS256 token sent to sms_gateway)
        - `secret` (string; secret used to sign the HS256 token sent to sms_gateway)
        - `secret_base64_encoded` (bool; indicates whether or not the secret is base64 encoded)
    - `forward_request_info` (bool; indicates whether or not request info should be forwarded to sms gateway)
- NOTE: In the request to configure the connection as a custom sms gateway, the following fields would also need to be updated:
    - `twilio_sid` (removed or set to "")
    - `twilio_token` (removed or set to "")
- NOTE: these fields would only be valid on "sms" connections

The full setup of a custom SMS gateway for passwordless is described in [this Auth0 post](https://auth0.com/docs/connections/passwordless/guides/use-sms-gateway-passwordless).

### New or Affected Resource(s)

* `auth0_connection` resource would require updates
* `gopkg.in/auth0.v5` will also require updates

### Potential Terraform Configuration

```hcl
resource "auth0_connection" "custom_sms_connection" {
  name     = "sms"
  strategy = "sms"
  ...
  options {
    provider    = "sms_gateway"
    gateway_url = "https://test.com/sms-gateway"
    gateway_authentication {
      method                = "bearer"
      subject               = "test.us.auth0.com:sms"
      audience              = "https://test.com/sms-gateway"
      secret                = "<My Shared Secret>"
      secret_base64_encoded = false
    }
    forward_request_info = true
    ...
  }
}
```

### References

- [Auth0 post: use-sms-gateway-passwordless](https://auth0.com/docs/connections/passwordless/guides/use-sms-gateway-passwordless)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by grese** [comment]
Just opened PR #417, which adds the resources described in this PR.

**Comment by tristan-curve** [comment]
> Just opened PR #417, and [go-auth0/auth0 #230](https://github.com/go-auth0/auth0/pull/230) which adds the options described in this PR to the `auth0_connection` resource.

what needs to happen to get this PR merged?

**Comment by grese** [comment]
@tristan-curve - Essentially, [go-auth0/auth0 #230](https://github.com/go-auth0/auth0/pull/230) just needs to be merged before #417.  It looks to me like both of the PRs still need review by @alexkappa. Also, if there are any requested changes during the review process, I'll just have to update the PRs.

**Comment by sergiughf** [comment]
Closing this as this will now be available in the next release as https://github.com/alexkappa/terraform-provider-auth0/pull/417 got merged. 

Thanks a lot for reporting this and opening the PR @grese !
